### PR TITLE
[opt](spill) add session variable of 'enable_force_spill'

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

We can set 'enable_force_spill' to ensure that the query always spills data to disk, regardless of available memory.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

File: be/src/pipeline/pipeline_fragment_context.cpp [hunk 1]
```diff
@@ -1167,7 +1167,8 @@ Status PipelineFragmentContext::_create_operator(ObjectPool* pool, const TPlanNo
             op.reset(new StreamingAggOperatorX(pool, next_operator_id(), tnode, descs));
             RETURN_IF_ERROR(cur_pipe->add_operator(op));
         } else {
-            if (_runtime_state->enable_agg_spill() && !tnode.agg_node.grouping_exprs.empty()) {
+            if ((_runtime_state->enable_force_spill() || _runtime_state->enable_agg_spill()) &&
```
Comment (yiguolei) [inline be/src/pipeline/pipeline_fragment_context.cpp:1170]: DO not modify like this.
in enable_agg_spill() method add a check enable_force_spill

File: regression-test/data/tpcds_sf1_p1/spill_test/q23.out [hunk 2]
```diff
@@ -0,0 +1,10 @@
+-- This file is automatically generated. You should know what you did if you want to edit this
```
Comment (yiguolei) [inline regression-test/data/tpcds_sf1_p1/spill_test/q23.out:1]: Does this test will run in PR pipeline?

File: be/src/runtime/runtime_state.h [hunk 3]
```diff
@@ -621,6 +621,10 @@ class RuntimeState {
         return _query_options.__isset.enable_agg_spill && _query_options.enable_agg_spill;
     }
 
+    bool enable_force_spill() const {
```
Comment (yiguolei) [inline be/src/runtime/runtime_state.h:624]: Do not need this method any more.

File: regression-test/suites/tpcds_sf1_p1/spill_test/q23.groovy [hunk 4]
```diff
@@ -0,0 +1,202 @@
+suite("q23") {
+    sql """set enable_force_spill=true"""
```
Comment (yiguolei) [inline regression-test/suites/tpcds_sf1_p1/spill_test/q23.groovy:2]: In this test case, we should set minRevocableMem explictly, to make sure spill disk is triggered. Not depend on fuzzy.
Then we could make sure, all PR will trigger for this test case.

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by jacktengg — APPROVED** [review]

**Pull request merged**